include/seqan3/range/view/join.hpp

```diff
@@ -0,0 +1,391 @@
+ */
+template <>
+constexpr bool add_enum_binary_ops<view_join_flags> = true;
+//!\endcond
+
+} // namespace seqan3
+
+namespace seqan3::detail
+{
+
+// --------------------------------------------------------------------------
+// FORWARDS
+// --------------------------------------------------------------------------
+
+// unconstrained forward declaration of the view
+template <typename irng_t, seqan3::view_join_flags flags>
+class view_join_ra;
+
+// unconstrained forward declaration of the iterator
+template <typename view_join_ra_t>
+class view_join_ra_iterator;
+
+// unconstrained forward declaration of the sentinel
+template <typename view_join_ra_t>
+class view_join_ra_sentinel;
+
+} // namespace seqan3::detail
+
+#include <seqan3/range/view/join_detail_eager.hpp>
+#include <seqan3/range/view/join_detail_lazy.hpp>
+
+namespace ranges::v3
+{
+
+template <typename irng_t, seqan3::view_join_flags flags>
+struct enable_view<seqan3::detail::view_join_ra<irng_t, flags>> : std::true_type
+{};
+
+} // namespace ranges::v3
+
+namespace seqan3::detail
+{
+
+// --------------------------------------------------------------------------
+// class join_fn (the generator type of view_join_ra)
+// --------------------------------------------------------------------------
+
+/*!\brief The type of seqan3::view::join_ra, a generator of seqan3::detail::view_join_ra.
+ * \ingroup view
+ */
+template <seqan3::view_join_flags flags>
+struct join_fn
+{
+    /*!\brief Forwards to ranges::view::join.
+     * \tparam irng_t The type of the range being processed. Required to be a seqan3::input_range_concept of
+     * seqan3::input_range_concept.
+     * \param irange The range being processed.
+     */
+    template <typename irng_t>
+    //!\cond
+        requires input_range_concept<irng_t> &&
+                 input_range_concept<ranges::range_reference_t<irng_t>>
+    //!\endcond
+    auto //view_join_ra<ranges::range_reference_t<ranges::range_reference_t<std::decay_t<irng_t>>>>
+    operator()(irng_t && irange) const
+    {
+        return irange | ranges::view::join;
+    }
+
+    /*!\brief Special overload for seqan3::concatenated_sequences that just calls
+     * seqan3::concatenated_sequences::concat().
+     * \tparam value_t Template parameters of seqan3::concatenated_sequences.
+     * \tparam delimiter_t Template parameters of seqan3::concatenated_sequences.
+     * \param irange The range being processed.
+     */
+    template <typename value_t, typename delimiter_t>
+    auto operator()(concatenated_sequences<value_t, delimiter_t> & irange) const
+    {
+        return irange.concat();
+    }
+
+    /*!\brief Special overload for seqan3::concatenated_sequences that just calls
+     * seqan3::concatenated_sequences::concat().
+     * \tparam value_t Template parameters of seqan3::concatenated_sequences.
+     * \tparam delimiter_t Template parameters of seqan3::concatenated_sequences.
+     * \param irange The range being processed.
+     */
```

Comment:
Can you mention in the documentation the difference (const) between the two overloads? It might be confusing otherwise as I would wonder why there are two overloads if the documentation is exactly the same.

Reply:
Well there is no difference in functionality, it's just that one takes const arguments (and return irange's const_reference type) and the other regular references (and return irange's reference type). But the generated doxygen show the parameter type as well...
But I can add it to the documentation.